### PR TITLE
DGC-137 Custom footer block

### PR DESCRIPTION
**Pull request by jonraedeke**
#137 

This is a start. @aweingarten there is placeholder data for footer link urls until we figure out the right method for that.

### CODE REVIEW

File: docroot/modules/custom/capitalcamp_blocks/src/Plugin/Block/DgcFooter.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  public function build() {
+    $parameters = new MenuTreeParameters();
+    $tree = $this->menuTree->load('footer', $parameters);
+    $processedFooter = [];
+
+    // Normalize the menu so that we don't need to in twig.
+    foreach ($tree as $treeElement) {
+      $currentParsedRecord = [];
+      $currentParsedRecord['text'] = $treeElement->link->getTitle();
+      $currentParsedRecord['url'] = 'https://www.drupalgovcon.org';
```
Comment (mikemadison13) [inline docroot/modules/custom/capitalcamp_blocks/src/Plugin/Block/DgcFooter.php]: @jonraedeke can you try replacing this line with the following and see if it works for you? it seems to work for me. if not, could you please give me some examples of what you have in your menu that "are not" working properly (and some that "are" working properly)?
`$currentParsedRecord['url'] = $treeElement->link->getUrlObject();`

**Comment by jonraedeke** [comment]
This footer block should be ready to go with config and even a behat test.

**Comment by sarahjean** [comment]
This looks great, I am going to go ahead and merge this. The footer links are still a placeholder right?

**Comment by jonraedeke** [comment]
@sarahjean Thanks so much. Footer links are dynamic thanks to @mikemadison13 .

**Pull request merged**